### PR TITLE
Enable RT722 EVB with MTL RVP on link0 single lane

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Early enabling of RT722 codec with MTL RVP on link 0 single lane. This patch
is verified with the topology https://github.com/thesofproject/sof/pull/8413.

I need more ramp up on enabling jack detection and codec init/exit, or some one could help to do it, but with this patch, headphone, speaker and DMIC works fine with single lane on link 0.

### CODE REVIEW

**Comment by bardliao** [comment]
For the init functions, you can refer to [sof_sdw_rt712_sdca.c](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw_rt712_sdca.c) and [sof_sdw_rt_sdca_jack_common.c](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/boards/sof_sdw_rt_sdca_jack_common.c)

File: sound/soc/intel/common/soc-acpi-intel-mtl-match.c [hunk 1]
```diff
@@ -135,6 +135,27 @@ static const struct snd_soc_acpi_endpoint rt712_endpoints[] = {
...
+		.group_position = 0,
+		.group_id = 0,
+	},
+	{
+		.num = 2,
+		.aggregated = 0,
+		.group_position = 0,
+		.group_id = 0,
+	},
+};
```
Comment (plbossart) [inline sound/soc/intel/common/soc-acpi-intel-mtl-match.c:161]: maybe add a comment on what those endpoints are. It's unusual to have multiple endpoints for a device. Same for the commit message.

File: sound/soc/intel/boards/sof_sdw.c [hunk 2]
```diff
@@ -870,6 +870,31 @@ static struct sof_sdw_codec_info codec_info_list[] = {
...
+				.direction = {true, true},
+				.dai_name = "rt722-sdca-aif1",
+				.dai_type = SOF_SDW_DAI_TYPE_JACK,
+				.dailink = {SDW_JACK_OUT_DAI_ID, SDW_JACK_IN_DAI_ID},
+			},
+			{
+				.direction = {true, false},
+				.dai_name = "rt722-sdca-aif2",
+				.dai_type = SOF_SDW_DAI_TYPE_AMP,
+				.dailink = {SDW_AMP_OUT_DAI_ID, SDW_UNUSED_DAI_ID},
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c:890]: are you sure there's no amp feedback for speaker protection? That's worthy of a comment.

File: sound/soc/intel/boards/sof_sdw.c [hunk 3]
```diff
@@ -870,6 +870,31 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 		},
 		.dai_num = 1,
 	},
+	{
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_sdw.c:873]: the commit message is a bit misleading.

Suggested edit:

"
RT722 is a multi-function codec which supports headset,
amp, and dmic functions. Each function provides a DAI which can be used in different dailinks.

The RT711 supports up to 3 SoundWire lanes, but that should not have any impact in the machine driver: the lanes are allocated and controlled by the manager and bandwidth allocation algorithm.
"

**Review by plbossart — CHANGES_REQUESTED** [review]

File: sound/soc/intel/boards/sof_sdw.c (hunk 2 above)
Comment (bardliao) [reply sound/soc/intel/boards/sof_sdw.c:890]: > are you sure there's no amp feedback for speaker protection? That's worthy of a comment.

The rt722-sdca codec driver doesn't provide capture capability for "rt722-sdca-aif2" dai. So, we can't create a capture dai link for it. And so does rt712-sdca.

**Comment by aiChaoSONG** [comment]
Update: 
- Add comments and revise commit message according to reviews
- add code init/exit implementation (thanks @bardliao for pointing out the rt712 reference code)

File: sound/soc/intel/boards/Makefile [hunk 4]
```diff
@@ -41,9 +41,9 @@ snd-soc-sof-sdw-objs += sof_sdw.o				\
 			sof_sdw_rt5682.o sof_sdw_rt700.o	\
 			sof_sdw_rt711.o sof_sdw_rt_sdca_jack_common.o	\
 			sof_sdw_rt712_sdca.o sof_sdw_rt715.o	\
-			sof_sdw_rt715_sdca.o sof_sdw_dmic.o	\
-			sof_sdw_cs42l42.o sof_sdw_cs42l43.o	\
-			sof_sdw_cs_amp.o			\
+			sof_sdw_rt715_sdca.o sof_sdw_rt722_sdca.o	\
+			sof_sdw_dmic.o sof_sdw_cs42l42.o	\
+			sof_sdw_cs42l43.o sof_sdw_cs_amp.o	\
```
Comment (plbossart) [inline sound/soc/intel/boards/Makefile]: best to keep some sort of order, e.g.
sof_sdw_rt715_sdca.o sof_sdw_rt722_sdca.o
sof_sdw_cs42l42.o sof_sdw_cs42l43.o	\
sof_sdw_cs_amp.o
sof_sdw_dmic.o
sof_sdw_hdmi.o

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Pull request merged**